jdaviz/configs/imviz/plugins/compass/compass.py

```diff
@@ -71,6 +71,9 @@ def draw_compass(self, data_label, img_data):
         Input is rendered buffer from Matplotlib.
 
         """
-        self.icon = self.app.state.layer_icons.get(data_label)
+        if self.app.loading or (icn := self.app.state.layer_icons.get(data_label)) is None:
+            return

```

Comment:
cool excuse to use `:=`, but this feels like a bandaid to an underlying problem.  Is this just because `draw_compass` is being called before the layer icon gets populated?  Would defaulting to empty string (so the `data_label` and `img_data` can update in the meantime work?  What ends up re-triggering another call to `draw_compass` so that it eventually gets refreshed?

Reply:
I don't have time to deep-dive into this one. Feel free to take over if you want or wait for it to come up again during sprint planning. This can always be a fallback fix if no better fix can be found.

Reply:
ok, I have no objection to getting this in as-is, but do think we should leave the ticket open for a deeper investigation since I worry this is just a symptom of a bigger problem.

Reply:
Okay. This PR won't close the issue. Not sure how to handle the JIRA one though.